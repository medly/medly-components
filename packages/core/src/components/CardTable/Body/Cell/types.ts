import { ColumnConfig, Data } from '../../types';

export interface Props {
    data: any;
    column: ColumnConfig;
    rowId: any;
    isRowClickDisabled?: boolean;
    rowData: Data;
}

export interface StyledProps {
    align?: 'left' | 'center' | 'right';
    wrapText?: boolean;
}
