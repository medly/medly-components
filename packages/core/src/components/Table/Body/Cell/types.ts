import { HTMLProps } from '@medly-components/utils';
import { ObjectType, TableColumnConfig, TableProps } from '../../types';

export type FreezePosition = 'left' | 'right';

export type TableCellStyledProps = {
    hidden?: boolean;
    frozen?: boolean;
    freezePosition?: FreezePosition;
    align?: 'left' | 'right' | 'center';
    wrapText?: boolean;
    tableSize: Required<TableProps>['size'];
};

export type TableCellProps = TableCellStyledProps &
    HTMLProps<HTMLTableCellElement> & {
        data: any;
        rowId: any;
        rowData: ObjectType;
        config: TableColumnConfig;
        dottedFieldName: string;
        isLoading?: boolean;
        isRowClickDisabled?: boolean;
        hiddenDivRef: React.RefObject<any>;
        tableSize?: TableProps['size'];
        addColumnMaxSize?: (field: string, value: number) => void;
        as?: keyof JSX.IntrinsicElements | React.ComponentType<any>;
    };
