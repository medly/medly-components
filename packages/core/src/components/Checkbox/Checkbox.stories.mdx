import { Checkbox } from './Checkbox';
import { Preview, Story, Meta, Props } from '@storybook/addon-docs/blocks';
import { labelPositions, sizes, ThemeInterface, FormWithCheckbox } from './Checkbox.stories';
import { useState } from 'react';
import { boolean, select, text } from '@storybook/addon-knobs';

<Meta
    title="Core"
    component={Checkbox}
    parameters={{
        jest: ['Checkbox.test.tsx'],
        design: {
            type: 'figma',
            url: 'https://www.figma.com/file/vA4EQdxl0d0ETtYpw2uSjW/Medly-Pattern-Library-B2B-Internal?node-id=239%3A0'
        }
    }}
/>

# Checkbox

It's a simple checkbox and a controlled react component. Handle the state as follows:

```tsx
const [isChecked, setChecked] = useState(false),
    onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => setChecked(e.target.checked);

return <Checkbox label="Lorem Ipsum" checked={isChecked} onChange={onChangeHandler} />;
```

<Preview withToolbar>
    <Story name="Checkbox">
        {() => {
            const [isChecked, setChecked] = useState(false),
                onChangeHandler = e => {
                    const val = e.target.checked;
                    setChecked(val);
                };
            return (
                <Checkbox
                    name="fruit"
                    size={select('Size', sizes, 'S')}
                    label={text('Label', 'A simple checkbox with label')}
                    disabled={boolean('Disabled', false)}
                    hasError={boolean('Error', false)}
                    required={boolean('Required', false)}
                    fullWidth={boolean('Full Width', false)}
                    indeterminate={boolean('Indeterminate', false)}
                    labelPosition={select('Label Position', labelPositions, 'right')}
                    errorText={text('Error Text', '')}
                    checked={isChecked}
                    onChange={onChangeHandler}
                />
            );
        }}
    </Story>
</Preview>

### Validations

1. This component calls the validator prop function with the latest value on blur, invalid, and change events. If you haven't passed the validator function, it will show the default html5 error validation message, if present.

```tsx
export const FormWithCheckbox: React.FC = props => {
    const [checked, setChecked] = useState<boolean>(),
        onChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => setChecked(e.target.checked),
        handleSubmit = (e: React.FormEvent) => e.preventDefault();
    return (
        <form onSubmit={handleSubmit}>
            <Checkbox required label="By clicking Submit, I agree lorem ipsum" checked={checked} onChange={onChangeHandler} {...props} />
            <Button type="submit">Submit</Button>
        </form>
    );
};
```

<Preview withToolbar>
    <FormWithCheckbox />
</Preview>

2. If you pass the validator function, it will be called with the latest selected value, so that, if required you can return the desired error message.


```tsx
validator={val => (!val ? 'Please accept the terms & conditions to continue.' : '')}
```

<Preview withToolbar>
    <FormWithCheckbox
        label="By clicking Submit, I agree lorem"
        validator={val => (!val ? 'Please accept the terms & conditions to continue.' : '')}
    />
</Preview>

### Props

<Props of={Checkbox} />

### CheckboxTheme

<Props of={ThemeInterface} />
