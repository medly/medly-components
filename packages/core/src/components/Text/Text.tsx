import { WithStyle } from '@medly-components/utils';
import React, { FC, useMemo } from 'react';
import { TextStyled } from './Text.styled';
import { TextProps } from './types';

export const Text: FC<TextProps> & WithStyle = React.memo(
    React.forwardRef(({ as, textVariant, textWeight, textAlign, children, ...restProps }, ref) => {
        const componentType = useMemo(() => {
            let type: keyof JSX.IntrinsicElements | React.ComponentType<any> = (textVariant as keyof JSX.IntrinsicElements) || 'span';

            if (
                !textVariant ||
                textVariant === 'body1' ||
                textVariant == 'body2' ||
                textVariant == 'body3' ||
                textVariant == 'button1' ||
                textVariant == 'button2'
            ) {
                type = textWeight === 'Strong' || textWeight === 'ExtraStrong' ? 'strong' : 'span';
            }

            return type;
        }, [textWeight, textVariant]);
        return (
            <TextStyled {...{ textVariant, textWeight, textAlign, ...restProps }} as={as || componentType} ref={ref}>
                {children}
            </TextStyled>
        );
    })
);

Text.defaultProps = {
    lineThrough: false,
    uppercase: false,
    fullWidth: false,
    textColor: 'inherit',
    textAlign: 'initial'
};
Text.displayName = 'Text';
Text.Style = TextStyled;
