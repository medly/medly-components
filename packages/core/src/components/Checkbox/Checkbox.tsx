import { CheckIcon, MinimizeIcon } from '@medly-components/icons';
import { useCombinedRefs, WithStyle } from '@medly-components/utils';
import React, { FC, FocusEvent, FormEvent, useCallback, useMemo, useState } from 'react';
import { SelectorLabel } from '../Selectors';
import * as Styled from './Checkbox.styled';
import { CheckboxProps } from './types';
const Component: FC<CheckboxProps> = React.memo(
    React.forwardRef((props, ref) => {
        const {
            id,
            size,
            fullWidth,
            label,
            labelPosition,
            labelVariant,
            labelWeight,
            indeterminate,
            validator,
            hasError,
            errorText,
            className,
            ...inputProps
        } = props;

        const [builtInErrorMessage, setErrorMessage] = useState(''),
            inputId = useMemo(() => id || label, [id, label]),
            inputRef = useCombinedRefs<HTMLInputElement>(ref, React.useRef(null)),
            isActive = useMemo(() => inputProps.checked || inputProps.defaultChecked || indeterminate, [
                inputProps.checked,
                inputProps.defaultChecked,
                indeterminate
            ]),
            isErrorPresent = useMemo(() => !!errorText || hasError || !!builtInErrorMessage, [errorText, hasError, builtInErrorMessage]);

        const validate = useCallback(
                (event: FormEvent<HTMLInputElement>, eventFunc?: any, preventDefault = true) => {
                    preventDefault && event.preventDefault();
                    const element = event.target as HTMLInputElement,
                        message = (validator && validator(element.checked)) || element.validationMessage;
                    setErrorMessage(message);
                    eventFunc && eventFunc(event);
                },
                [validator]
            ),
            onBlur = useCallback((event: FocusEvent<HTMLInputElement>) => validate(event, props.onBlur), [validate, props.onBlur]),
            onInvalid = useCallback((event: FormEvent<HTMLInputElement>) => validate(event, props.onInvalid), [validate, props.onInvalid]),
            onChange = useCallback((event: FormEvent<HTMLInputElement>) => validate(event, props.onChange, false), [
                validate,
                props.onChange
            ]);

        return (
            <>
                {(!!errorText || builtInErrorMessage) && (
                    <Styled.ErrorText disabled={inputProps.disabled}>{errorText || builtInErrorMessage}</Styled.ErrorText>
                )}
                <Styled.CheckboxWithLabelWrapper
                    className={className}
                    id={`${inputId}-wrapper`}
                    htmlFor={inputId}
                    isActive={isActive}
                    hasError={isErrorPresent}
                    disabled={inputProps.disabled}
                    {...{ fullWidth, labelPosition }}
                >
                    {label && (
                        <SelectorLabel
                            id={`${inputId}-label`}
                            type="checkbox"
                            disabled={inputProps.disabled}
                            hasError={isErrorPresent}
                            labelPosition={labelPosition}
                            textVariant={labelVariant}
                            textWeight={labelWeight}
                            required={inputProps.required}
                        >
                            {label}
                        </SelectorLabel>
                    )}
                    <Styled.CheckboxWrapper size={size}>
                        <Styled.HiddenCheckbox
                            ref={inputRef}
                            id={inputId}
                            indeterminate={indeterminate}
                            hasError={isErrorPresent}
                            {...{ ...inputProps, onBlur, onInvalid, onChange }}
                        />
                        <Styled.StyledCheckbox>{indeterminate ? <MinimizeIcon /> : <CheckIcon />}</Styled.StyledCheckbox>
                    </Styled.CheckboxWrapper>
                </Styled.CheckboxWithLabelWrapper>
            </>
        );
    })
);

Component.displayName = 'Checkbox';
Component.defaultProps = { label: '', labelPosition: 'right' };
export const Checkbox: FC<CheckboxProps> & WithStyle = Object.assign(Component, { Style: Styled.CheckboxWithLabelWrapper });
