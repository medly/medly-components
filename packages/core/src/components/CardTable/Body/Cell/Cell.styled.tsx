import styled, { css } from 'styled-components';
import Text from '../../../Text';
import { StyledProps } from './types';

const wrapTextStyle = () => css`
    ${Text.Style} {
        text-overflow: ellipsis;
        overflow: hidden;
        white-space: nowrap;
    }
`;

export const Cell = styled('td')<StyledProps>`
    padding: 0;
    padding: 1.5rem 2.4rem;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: ${({ align }) => (align === 'left' ? 'flex-start' : align === 'right' ? 'flex-end' : 'center')};
    overflow: hidden;

    ${({ wrapText }) => wrapText && wrapTextStyle()}
`;

Cell.defaultProps = {
    align: 'left',
    wrapText: true
};
