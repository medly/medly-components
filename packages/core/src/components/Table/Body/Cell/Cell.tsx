import { WithStyle } from '@medly-components/utils';
import React, { useEffect, useMemo, useRef } from 'react';
import Text from '../../../Text';
import { Cell as StyledCell, CustomComponentWrapper, LoadingDiv } from './Styled';
import { TableCellProps } from './types';

const Cell: React.FC<TableCellProps> & WithStyle = React.memo(props => {
    const childRef = useRef(null),
        { addColumnMaxSize, config, data, rowId, rowData, isRowClickDisabled, dottedFieldName, tableSize, isLoading, ...restProps } = props,
        { align, hidden, frozen, formatter, component: CustomComponent } = config,
        formattedData = useMemo(() => (formatter ? formatter(data) : data), [formatter, data]);

    useEffect(() => {
        childRef.current &&
            !isLoading &&
            addColumnMaxSize &&
            addColumnMaxSize(dottedFieldName, childRef.current.clientWidth + (tableSize === 'L' ? 48 : 32));
    }, [childRef, tableSize]);

    return (
        <StyledCell hidden={hidden} frozen={frozen} tableSize={tableSize} align={align} {...restProps}>
            {isLoading ? (
                <LoadingDiv ref={childRef} />
            ) : CustomComponent ? (
                <CustomComponentWrapper ref={childRef}>
                    <CustomComponent {...{ data: formattedData, rowId, disabled: isRowClickDisabled, rowData }} />
                </CustomComponentWrapper>
            ) : (
                <Text ref={childRef} textVariant="body2" title={formattedData}>
                    {formattedData}
                </Text>
            )}
        </StyledCell>
    );
});
Cell.displayName = 'Cell';
Cell.Style = StyledCell;

export default Cell;
