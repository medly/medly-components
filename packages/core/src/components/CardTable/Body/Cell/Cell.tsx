import React, { useMemo } from 'react';
import Text from '../../../Text';
import * as Style from './Cell.styled';
import { Props } from './types';

export const Cell: React.FC<Props> = React.memo(props => {
    const { data, rowId, column, isRowClickDisabled, rowData } = props,
        { component: Component, align, wrapText } = column;

    const FormattedCell = useMemo(
        () => (Component ? <Component {...{ data, rowId, rowData, disabled: isRowClickDisabled }} /> : <Text title={data}>{data}</Text>),
        [data, rowId, isRowClickDisabled, Component]
    );

    return (
        <Style.Cell wrapText={wrapText} align={align}>
            {FormattedCell}
        </Style.Cell>
    );
});
Cell.displayName = 'Cell';
