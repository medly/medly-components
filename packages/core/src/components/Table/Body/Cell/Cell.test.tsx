import { render } from '@test-utils';
import React from 'react';
import Checkbox from '../../../Checkbox';
import Text from '../../../Text';
import TableCell from './Cell';
import { Cell } from './Styled/Cell.styled';

describe('Cell', () => {
    it('should render text properly', () => {
        const { container } = render(
            <Cell tableSize="M">
                <Text>Dummy</Text>
            </Cell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render frozen text properly', () => {
        const { container } = render(
            <Cell frozen tableSize="M">
                <Text>Dummy</Text>
            </Cell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should not render text if hidden property is given', () => {
        const { container } = render(
            <Cell hidden tableSize="M">
                <Text>Dummy</Text>
            </Cell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render checkbox properly', () => {
        const { container } = render(
            <Cell tableSize="M">
                <Checkbox label="Dummy" />
            </Cell>
        );
        expect(container).toMatchSnapshot();
    });

    it('should render tooltip properly', () => {
        const { container } = render(
            <TableCell
                data="hello"
                rowId="1"
                rowData={{}}
                tableSize="M"
                hiddenDivRef={{ current: jest.fn() }}
                dottedFieldName="title"
                config={{ title: 'Hello', field: 'hello' }}
            />
        );
        expect(container).toMatchSnapshot();
    });
});
