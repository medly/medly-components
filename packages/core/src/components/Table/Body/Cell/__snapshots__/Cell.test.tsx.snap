// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell should not render text if hidden property is given 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 0;
  padding: 0;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 .c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<div>
  <td
    class="c0"
    hidden=""
  >
    <span
      class="c1 c2"
    >
      Dummy
    </span>
  </td>
</div>
`;

exports[`Cell should render checkbox properly 1`] = `
.c10 {
  overflow: visible;
  font-size: 2.4rem;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  cursor: inherit;
}

.c10 * {
  fill-opacity: 1;
  -webkit-transition: all 100ms linear;
  transition: all 100ms linear;
  fill: #607890;
}

.c3 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c4 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: inherit;
  color: inherit;
  padding: 0 1.6rem;
}

.c8 {
  border: 0.15rem solid;
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  border-radius: 22.2%;
  -webkit-transition: all 100ms ease-out;
  transition: all 100ms ease-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 .c9 {
  pointer-events: none;
  z-index: 1;
  -webkit-transition: all 200ms ease-in-out;
  transition: all 200ms ease-in-out;
  -webkit-transform: scale(0);
  -ms-transform: scale(0);
  transform: scale(0);
  width: 100%;
  height: 100%;
  margin-right: 0.05rem;
}

.c6 {
  opacity: 0;
  margin: 0;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  outline: none;
  position: absolute;
}

.c6:checked ~ .c7 {
  border-color: #126AFA;
  background-color: #126AFA;
}

.c6:checked ~ .c7 .c9 {
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c6:checked ~ .c7 .c9 * {
  fill: #ffffff;
}

.c6:not(:checked) ~ .c7 {
  background-color: transparent;
  border-color: #13181D;
}

.c6:not(:disabled):focus ~ .c7 {
  box-shadow: 0 0 0.8rem 0 rgba(18,106,250,0.35);
  border-color: #126AFA;
}

.c6:not(:disabled):focus:not(:checked) ~ .c7 {
  border-color: #126AFA;
}

.c5 {
  margin: 0.3rem;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 1.8rem;
  height: 1.8rem;
  position: relative;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  padding: 0.8rem 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c1 * {
  cursor: pointer;
}

.c1.c1.c1:hover .c7 {
  box-shadow: 0 0 0.8rem 0 rgba(18,106,250,0.35);
  border-color: #126AFA;
}

.c1.c1.c1:active .c7 {
  box-shadow: 0 0 0.8rem 0 rgba(18,106,250,0.5);
  border-color: #0F5AD5;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 .c2 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<div>
  <td
    class="c0"
  >
    
    <label
      class="c1"
      for="Dummy"
      id="Dummy-wrapper"
      sizes="[object Object]"
    >
      <span
        class="c2 c3 c4"
        id="Dummy-label"
        type="checkbox"
      >
        Dummy
      </span>
      <div
        class="c5"
      >
        <input
          class="c6"
          id="Dummy"
          sizes="[object Object]"
          type="checkbox"
        />
        <div
          class="c7 c8"
        >
          <svg
            class="c9 c10"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.795 15.875l-3.47-3.47a.996.996 0 0 0-1.41 0 .996.996 0 0 0 0 1.41l4.18 4.18c.39.39 1.02.39 1.41 0l10.58-10.58a.996.996 0 1 0-1.41-1.41l-9.88 9.87z"
              fill="green"
            />
          </svg>
        </div>
      </div>
    </label>
  </td>
</div>
`;

exports[`Cell should render frozen text properly 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  position: -webkit-sticky;
  position: sticky;
  left: 0;
  z-index: 1;
}

.c0 * {
  z-index: 1;
}

.c0 .c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<div>
  <td
    class="c0"
  >
    <span
      class="c1 c2"
    >
      Dummy
    </span>
  </td>
</div>
`;

exports[`Cell should render text properly 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 .c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<div>
  <td
    class="c0"
  >
    <span
      class="c1 c2"
    >
      Dummy
    </span>
  </td>
</div>
`;

exports[`Cell should render tooltip properly 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 1.4rem;
  font-weight: 400;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  line-height: 2.2rem;
  text-align: initial;
}

.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  opacity: 1;
  padding: 1.2rem 1.6rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c0 .c1 {
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
}

<div>
  <td
    class="c0"
  >
    <span
      class="c1 c2"
      title="hello"
    >
      hello
    </span>
  </td>
</div>
`;
